morphocluster/api.py

```diff
@@ -232,6 +235,81 @@ def get_file(path):
     )
 
 
+@api.route("/files/<path:path>", methods=["GET"])
+@api.route("/files/", methods=["GET"])
+def get_direntry(path=""):
+
+    directory_path = os.path.join(app.config["FILES_DIR"],path)
+
+    file_list = []
+
+    for entry in os.scandir(directory_path):
+        entry_info = {
+            "name": entry.name,
+            "path": os.path.relpath(entry.path, api.config["FILES_DIR"]),
+            "type": "directory" if entry.is_dir() else "file",
+            "last_modified": datetime.fromtimestamp(entry.stat().st_mtime).isoformat(),
+            "type_flag": 1 if entry.is_dir() else 0,
+        }
+        file_list.append(entry_info)
+
+    return jsonify(file_list)
+
+@api.route("/file/info/<path:file_path>", methods=["GET"])
+def get_file_info(file_path):
+    file_list = []
+    try:
+
+        file_path_on_server = os.path.join(app.config["FILES_DIR"], file_path)
+
+        if os.path.exists(file_path_on_server) and os.path.isfile(file_path_on_server):
```

Comment:
`exists(...) and isfile(...)` is redundant. `isfile` is enough. Also, return early if file not found:

```python
if not os.path.isfile(path):
    raise werkzeug.exceptions.NotFound()

file_info = {...}
return jsonify(file_info)
```

Also, don't create a list, you only return information on one file!